### PR TITLE
Add provider default labels

### PR DESCRIPTION
**Pull request by zli82016**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by c2thorn** [comment]
Noting that this fixes https://github.com/hashicorp/terraform-provider-google/issues/7325

**Review by rileykarson — APPROVED** [review]

File: mmv1/third_party/terraform/fwprovider/framework_provider.go.erb [hunk 1]
```diff
@@ -113,6 +113,10 @@ func (p *FrameworkProvider) Schema(_ context.Context, _ provider.SchemaRequest,
             "request_reason": schema.StringAttribute{
                 Optional: true,
             },
+            "default_labels": schema.MapAttribute{
+				Optional:    true,
```
[the remark this answers is not in the document]
Comment (zli82016) [reply mmv1/third_party/terraform/fwprovider/framework_provider.go.erb:117]: Good catch.

**Review by melinath — CHANGES_REQUESTED** [review]
Regarding the TGC failures - it looks like this PR renames `labels` fields to `terraform_labels`; if that's correct, I believe the fix here would be to modify the [TGC test files for impacted resources](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/tgc/tests/data) so that:

- the `.tf` file uses `terraform_labels`
- the `.tfplan.json` file reflects the `terraform_labels` name as well. This should just be a string replacement but you could copy the .tf file to a new directory and run `terraform plan` to verify if there are any issues.

Once you've made those changes, the expander should be able to automatically convert the `terraform_labels` value to a `labels` value in the CAI resource (which means it should match the existing expected CAI output.)

**Comment by zli82016** [comment]
> Regarding the TGC failures - it looks like this PR renames `labels` fields to `terraform_labels`; if that's correct, I believe the fix here would be to modify the [TGC test files for impacted resources](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/tgc/tests/data) so that:
> 
> * the `.tf` file uses `terraform_labels`
> * the `.tfplan.json` file reflects the `terraform_labels` name as well. This should just be a string replacement but you could copy the .tf file to a new directory and run `terraform plan` to verify if there are any issues.
> 
> Once you've made those changes, the expander should be able to automatically convert the `terraform_labels` value to a `labels` value in the CAI resource (which means it should match the existing expected CAI output.)

For the users, only `labels` field is provided in the configuration. `terraform_labels` will be added to the terraform plan by the provider.

**Comment by melinath** [comment]
hm, in that case the configurations should stay the same. It would be good to see what you get when you generate the `tfplan.json` files - the process for generating that file is described at https://github.com/GoogleCloudPlatform/terraform-google-conversion/blob/main/docs/contributing/add_new_resource.md#2-adding-new-tests

**Comment by melinath** [comment]
Hopefully the `tfplan.json` files will automatically have `terraform_labels` present; if not, this will be more complex to fix and I'll need to take a closer look.

**Comment by zli82016** [comment]
I haven't try the process in the link https://github.com/GoogleCloudPlatform/terraform-google-conversion/blob/main/docs/contributing/add_new_resource.md#2-adding-new-tests

But when I generate tgc locally with the command `make tgc OUTPUT_PATH=~/Documents/workspace/terraform-google-conversion/tfplan2cai`. There are not `terraform_labels` field in [tfplan.json](https://paste.googleplex.com/4546126521827328) file.

**Comment by melinath** [comment]
please try the process I linked to update the files in Magic Modules - that needs to happen for downstream generation to work properly.

File: mmv1/templates/terraform/examples/base_configs/test_file.go.erb [hunk 2]
```diff
@@ -44,7 +44,7 @@ object.examples
     test_slug = "#{resource_name}_#{example.name.camelize(:lower)}Example"
 
     ignore_read = object.all_user_properties
-      .select{|p| p.url_param_only || p.ignore_read || p.is_a?(Api::Type::ResourceRef) || p.is_a?(Api::Type::KeyValueLabels || p.is_a?(Api::Type::KeyValueAnnotations))}
+      .select{|p| p.url_param_only || p.ignore_read || p.is_a?(Api::Type::ResourceRef) || p.is_a?(Api::Type::KeyValueLabels) || p.is_a?(Api::Type::KeyValueAnnotations) || p.name == 'terraformLabels'}
```
[the remark this answers is not in the document]
Comment (zli82016) [reply mmv1/templates/terraform/examples/base_configs/test_file.go.erb:47]: Yeah, the new type is better. I will add a new type `KeyValueTerraformLabels `

**Comment by zli82016** [comment]
Rebased the 5.0.0 branch to address conflicts.

**Comment by rileykarson** [comment]
Is the issue with `terraform_labels` or with `labels`? I see the TGC failures looking for a `labels` value and not finding it.

**Comment by zli82016** [comment]
> Is the issue with `terraform_labels` or with `labels`? I see the TGC failures looking for a `labels` value and not finding it.



> Is the issue with `terraform_labels` or with `labels`? I see the TGC failures looking for a `labels` value and not finding it.

This is the new generated bigquery dataset converter file. https://github.com/modular-magician/terraform-google-conversion/blob/9b1045201b1d732ab4740388ace5844fb210efb7/tfplan2cai/converters/google/resources/services/bigquery/bigquery_dataset.go#L159

The value `labels` is from expanding `terraform_labels`. Based on my understanding, the schema in the google-beta provider used in tgc doesn't have `terraform_labels` field yet, so the generated fake resource data (d) doesn't have `terraform_labels` field.

This issue could be fixed after merged back to the main branch.

**Comment by zli82016** [comment]
I can remove the `labels` field from the testdata files to not break tgc. And then check if the issue still persists after merging back to the main branch.

**Review by melinath — CHANGES_REQUESTED** [review]
I've looked into this a little more closely & tried to regenerate the test files to see if that resolves the issue. The test generation docs are woefully out of date and need to be updated.

However, I was able to confirm that:

1. `terraform_labels` should be set in the tfplan.json files
2. If `terraform_labels` is set in the tfplan.json files, the converted CAI will include the correct `labels` value.

So the fix here is: For all [`*.tfplan.json` files for impacted resources](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/tgc/tests/data), duplicate the existing `labels` value as `terraform_labels`. That should resolve the issue.

**Comment by zli82016** [comment]
> Details

Hello, @melinath , thanks for the comments. I think that is what I tried before. `terraform-google-conversion-test-integration-0.12.31` and  `terraform-google-conversion-test-integration-0.13.7` still failed.

**Comment by zli82016** [comment]
I think  [the previous comment](https://github.com/GoogleCloudPlatform/magic-modules/pull/8670#issuecomment-1692396646) could be reason.

**Comment by melinath** [comment]
It looks to me like this ought to be passing at this point. The unit tests are passing; if this were a fundamental issue then I would expect them to also be failing. I'm going to try rerunning tests to see if that resolves the issue.

/gcbrun

**Comment by zli82016** [comment]
terraform-google-conversion-test-integration-0.12.31 and terraform-google-conversion-test-integration-0.13.7 still failed. Close and reopen this PR to see if the tests will pass.

**Review by melinath — APPROVED** [review]

**Review by rileykarson — CHANGES_REQUESTED** [review]
Marking requested as I'll make a final pass after some of the changes

**Review by rileykarson — APPROVED** [review]

**Pull request merged**